apps/mobile/src/components/OfferScreen/atoms/editOfferAtoms.ts

```diff
@@ -0,0 +1,98 @@
+import {atom, type SetStateAction, type WritableAtom} from 'jotai'
+import {
+  type ConnectionLevel,
+  OfferId,
+  SymmetricKey,
+} from '@vexl-next/domain/dist/general/offers'
+import {createScope, molecule} from 'jotai-molecules'
+import {type OneOfferInState} from '../../../state/marketplace/domain'
+import {OfferAdminId} from '@vexl-next/rest-api/dist/services/offer/contracts'
+import {IdNumeric} from '@vexl-next/domain/dist/utility/IdNumeric'
+import {IsoDatetimeString} from '@vexl-next/domain/dist/utility/IsoDatetimeString.brand'
+import {PublicKeyPemBase64} from '@vexl-next/cryptography/dist/KeyHolder'
+import {createOfferStateAtom} from './createOfferAtoms'
+import valueOrDefaultAtom from '../../../utils/atomUtils/valueOrDefaultAtom'
+import {singleOfferAtom} from '../../../state/marketplace/atom'
+
+export const dummyOffer: OneOfferInState = {
+  adminId: OfferAdminId.parse('offerAdminId'),
+  flags: {
+    isMine: true,
+    reported: false,
+    intendedConnectionLevel: 'ALL',
+  },
+  offerInfo: {
+    id: IdNumeric.parse(1),
+    offerId: OfferId.parse('offerId'),
+    privatePart: {
+      commonFriends: ['Mike', 'John'],
+      friendLevel: ['FIRST_DEGREE'],
+      symmetricKey: SymmetricKey.parse('symmetricKey'),
+    },
+    publicPart: {
+      offerPublicKey: PublicKeyPemBase64.parse('offerPublicKey'),
+      location: [],
+      offerDescription: '',
+      amountBottomLimit: 0,
+      amountTopLimit: 0,
+      feeState: 'WITHOUT_FEE',
+      feeAmount: 0,
+      locationState: 'IN_PERSON',
+      paymentMethod: ['CASH'],
+      btcNetwork: ['LIGHTING'],
+      currency: 'CZK',
+      offerType: 'SELL',
+      activePriceState: 'NONE',
+      activePriceValue: 0,
+      activePriceCurrency: 'CZK',
+      active: true,
+      groupUuids: [],
+    },
+    createdAt: IsoDatetimeString.parse('1970-01-01T00:00:00.000Z'),
+    modifiedAt: IsoDatetimeString.parse('1970-01-01T00:00:00.000Z'),
+  },
+}
+
+export const nonNullOfferAtom = (
+  offerId: OfferId
+): WritableAtom<
+  NonNullable<OneOfferInState>,
+  [SetStateAction<NonNullable<OneOfferInState>>],
+  void
+> =>
+  valueOrDefaultAtom({
+    nullableAtom: singleOfferAtom(offerId),
+    dummyValue: dummyOffer,
+  })
+
+export const OfferScope = createScope<
+  WritableAtom<OneOfferInState, [SetStateAction<OneOfferInState>], void>
+>(atom<OneOfferInState>(dummyOffer))
+
+export const offerMolecule = molecule((getMolecule, getScope) => {
```

Comment:
Is molecule needed here. It returns atoms that modify other atoms that are not part of this molecule